regression/cbmc/simplify-overflow/main.c

```diff
@@ -0,0 +1,9 @@
+int main()
+{
+  __CPROVER_assert(!__CPROVER_overflow_plus(1, 2), "");
+  __CPROVER_assert(__CPROVER_overflow_minus(1U, 2U), "");
+  __CPROVER_assert(!__CPROVER_overflow_mult(1U, 2U), "");
+  __CPROVER_assert(!__CPROVER_overflow_shl(1, 2), "");
+  __CPROVER_assert(!__CPROVER_overflow_unary_minus(1), "");
+  __CPROVER_assert(__CPROVER_overflow_unary_minus(1U), "");
+}
```

Comment:
Maybe we should also have:
__CPROVER_assert(__CPROVER_overflow_minus(0U, 2U), "");
To check the case below where a 0 value can still have overflow?

Reply:
If so, it should probably be in a separate test case as the trick @tautschnig is using for testing that it is simplification that handles these will not work for false things.

```
while (__CPROVER_overflow_minus(0U, 2U) {}
```

could do something interesting though...

Reply:
Trying to cover this in the test spec: `^Generated 7 VCC\(s\), 0 remaining after simplification$`

Reply:
I would say that this is both frail and also obscure.
How about using `__builtin_constant_p`?

Reply:
Fair. I'm now using `_Static_assert` instead.